src/sessions/__tests__/PageManager.test.ts

```diff
@@ -487,3 +488,86 @@ describe('PageManager tests', () => {
         expect(pageManager.getPage().start).toEqual(2500);
     });
 });
+
+test('when complete referrer is available from the DOM', async () => {
+    // Init
+    const config: Config = {
+        ...DEFAULT_CONFIG,
+        allowCookies: true
+    };
+    const pageManager: PageManager = new PageManager(config, record);
+
+    Object.defineProperty(document, 'referrer', {
+        value: 'http://abc.com/consoles',
+        configurable: true
+    });
+
+    // Run
+    pageManager.recordPageView('/console/home');
+
+    // Assert
+    expect(pageManager.getPage()).toMatchObject({
+        referrer: 'http://abc.com/consoles',
+        referrerDomain: 'http://abc.com',
+        pageId: '/console/home'
+    });
+
+    window.removeEventListener(
+        'popstate',
+        (pageManager as any).popstateListener
+    );
+});
+
+test('when only domain level referrer is available from the DOM', async () => {
```

Comment:
suggestion: `when only domain level referrer is available from the DOM` --> `when only domain level referrer is available from the DOM then is recorded in page view event`